### ISSUE TITLE
unused parameters in IRCClient.register

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11155]
|<img alt="synapsis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| synapsis reported|
|-|-|
|Trac ID|trac#1649|
|Type|defect|
|Created|2006-04-17 15:38:03Z|

The method register has two parameters
hostname='foo', servername='bar'

that are never used

<details><summary>Searchable metadata</summary>

```
trac-id__1649 1649
type__defect defect
reporter__synapsis synapsis
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1145288283000000 1145288283000000
changetime__1169057749000000 1169057749000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Fixed in ircclient-register-1649

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Looks good, great test coverage.

Probably unrelated, but is there a ticket opened to refactor some tests utilities ? It's bad to redefine StringIOWithoutClosing so many times.

Anyway please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [19378]) Merge ircclient-register-1649

Author: exarkun
Reviewer: therve
Fixes [#11155](https://github.com/twisted/twisted/issues/11155)

Respect the hostname and servername parameters to IRCClient.register and
add test coverage for that, as well as the connection password-based
authentication that method can also do.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
